### PR TITLE
CATROID-862 Incorrect elimination of "f" and "d" behind numbers in strings

### PR DESCRIPTION
**Pull request by Rawbat**
Fixed this by adding a check if those chars are contained in the string in the tryParseDouble Function.
Also removed the normal double cast in the set variable action and replaced it with a function call to this.
Also added a CLT to verify this behaviour and removed rtl test bug caused by CATROID-835.

https://jira.catrob.at/browse/CATROID-862

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/formulaeditor/common/Conversions.kt [hunk 1]
```diff
@@ -32,6 +32,9 @@ object Conversions {
     private val colorPattern = Pattern.compile("#\\p{XDigit}{6}")
 
     private fun tryParseDouble(argument: String): Double? {
+        if (argument.contains('f') || argument.contains('d')) {
+            return null
+        }
```
Comment (Koell) [inline catroid/src/main/java/org/catrobat/catroid/formulaeditor/common/Conversions.kt:37]: please replace the if with 
`if (argument.matches(Regex("[fFdD]\$"))){`
as currently the usecase ("F") which promptet this task is ignored

**Review by Koell — CHANGES_REQUESTED** [review]
currently the CTL would also wrongfully work for floats or double ending with upper "F/D"
direkt comparisons cast the variables to double as possible which is the correct behaviour, to check if the float is still stored as a float regexes work quiete nicely

**Review by Koell — APPROVED** [review]

**Comment by Koell** [comment]
Thanks for the work

**Review by wslany — CHANGES_REQUESTED** [review]
Nice! Please can you fix the lint warning? 

=> https://jenkins.catrob.at/job/Catroid/job/PR-3974/13/analysis/source.d38fecc7-1338-48fd-9893-2c094f2bd35f/#35

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!

**Pull request merged**